tests/e2e/specs/layers.spec.js

```diff
@@ -300,6 +310,70 @@ describe('Test of layer handling', () => {
                 expect(visibleIds).to.contain(testLayerId)
             })
         })
+        it('keeps the layer settings when changing language', () => {
+            const langBefore = 'en'
+            const langAfter = 'de'
+            let activeLayersConfigBefore
+
+            // Change the layer configurations to non-default values.
+            cy.readWindowValue('store').then((store) => {
+                visibleLayerIds.forEach((layerId) => {
+                    store.dispatch('toggleLayerVisibility', { layerId })
+                    store.dispatch('setLayerOpacity', {
+                        layerId,
+                        opacity: 0.1,
+                    })
+                })
+                cy.fixture('layers.fixture.json').then((fixtures) => {
+                    const layerId = 'test.timeenabled.wmts.layer'
+                    const layerMetadata = fixtures[layerId]
+                    const layerTimestamp = getRandomTimestampFromSeries(layerMetadata)
+                    store.dispatch('setTimedLayerCurrentTimestamp', {
+                        layerId,
+                        timestamp: layerTimestamp,
+                    })
+                })
+            })
+
+            cy.readStoreValue('state').then((state) => {
+                // Check the language before the switch.
+                expect(state.i18n.lang).to.eq(langBefore)
+                state.layers.activeLayers
+                    .filter((layer) => 'lang' in layer)
+                    .forEach((layer) => expect(layer.lang).to.eq(langBefore))
+                // Save the layer configuration before the switch.
+                activeLayersConfigBefore = JSON.stringify(
+                    state.layers.activeLayers,
+                    stringifyWithoutLang
+                )
+            })
+
+            // Wait until the layers have been updated. We have to wait here
+            // to make sure the next time we wait we catch the language change.
+            cy.wait('@layers-configured')
+
+            // Change the language to a non-default value.
+            cy.get('[data-cy="menu-settings-section"]').click()
+            cy.get(`#menu-lang-selector button:contains("${langAfter.toUpperCase()}")`).click()
```

Comment:
would it be possible to use a custom HTML attribute here (`data-cy-...`) instead of a normal CSS selector? That's a best practice with Cypress, so that we are not depending on the HTML/CSS styling couple, and clearly state in the HTML that this element will be used in further tests.

Reply:
Yeah. That was messy of me. I added the data-attributes and updated the selector in the test.